bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -15,100 +18,159 @@
 
 
 class Index(TemplateView):
-    ''' The homepage '''
-    template_name = 'recordtransfer/home.html'
+    """The homepage"""
+
+    template_name = "recordtransfer/home.html"
 
 
 class TransferSent(TemplateView):
-    ''' The page a user sees when they finish a transfer '''
-    template_name = 'recordtransfer/transfersent.html'
+    """The page a user sees when they finish a transfer"""
+
+    template_name = "recordtransfer/transfersent.html"
 
 
 class SystemErrorPage(TemplateView):
-    """ The page a user sees when there is some system error. """
-    template_name = 'recordtransfer/systemerror.html'
+    """The page a user sees when there is some system error."""
+
+    template_name = "recordtransfer/systemerror.html"
 
 
 class UserProfile(UpdateView):
-    ''' This view shows two things:
+    """View to show two things:
     - The user's profile information
-    - A list of the Submissions a user has created via transfer
-    '''
+    - A list of the Submissions a user has created via transfer.
+    """
 
-    template_name = 'recordtransfer/profile.html'
+    template_name = "recordtransfer/profile.html"
     paginate_by = 10
     form_class = UserProfileForm
-    success_url = reverse_lazy('recordtransfer:userprofile')
+    success_url = reverse_lazy("recordtransfer:userprofile")
 
     def get_object(self, queryset=None):
         return self.request.user
 
     def get_context_data(self, **kwargs):
         context = super(UserProfile, self).get_context_data(**kwargs)
-        context['in_process_submissions'] = SavedTransfer.objects.filter(user=self.request.user)\
-            .order_by('-last_updated')
-        context['user_submissions'] = Submission.objects.filter(user=self.request.user).order_by('-submission_date')
+        context["in_process_submissions"] = SavedTransfer.objects.filter(
+            user=self.request.user
+        ).order_by("-last_updated")
+        context["user_submissions"] = Submission.objects.filter(user=self.request.user).order_by(
+            "-submission_date"
+        )
+
+        context["ID_GETS_NOTIFICATION_EMAILS"] = ID_GETS_NOTIFICATION_EMAILS
+        context["ID_CURRENT_PASSWORD"] = ID_CURRENT_PASSWORD
+        context["ID_NEW_PASSWORD"] = ID_NEW_PASSWORD
+        context["ID_CONFIRM_NEW_PASSWORD"] = ID_CONFIRM_NEW_PASSWORD
         return context
 
+    def get_form_kwargs(self):
+        """Pass User instance to form as a keyword argument."""
+        kwargs = super(UserProfile, self).get_form_kwargs()
+        kwargs["user"] = self.get_object()
+        return kwargs
+
     def form_valid(self, form):
-        messages.success(self.request, 'Preferences updated')
-        return super().form_valid(form)
+        response = super().form_valid(form)
+        success_message = "Preferences updated"
+
+        if form.cleaned_data.get("new_password"):
+            update_session_auth_hash(self.request, form.instance)
+            success_message = "Password updated"
```

Comment:
"Password updated" should be surrounded by `gettext()`.

Reply:
Resolved by https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/208/commits/fd426e65267fdd2f018dc0cc8f56073b7b0271ec